integration/vite-css-dev-express-test.ts

```diff
@@ -48,15 +43,28 @@ test.describe("Vite CSS dev (Express server)", () => {
           import { createRequestHandler } from "@remix-run/express";
           import { installGlobals } from "@remix-run/node";
           import express from "express";
+          import http from "node:http";
+          import { createServer } from "vite";
 
           installGlobals();
 
+          const app = express();
+
+          const server = http.createServer(app);
+
+          // TODO: should support custom server config via "unstable_createViteServer"?
+          // https://github.com/sapphi-red/vite-setup-catalogue/blob/7ff3dd0850e5f575a01e222342f979bd70f46523/examples/middleware-mode/server.js#L12-L20
           let vite =
             process.env.NODE_ENV === "production"
-              ? undefined
-              : await unstable_createViteServer();
-
-          const app = express();
+            ? undefined
+            : await createServer({
+                server: {
+                  middlewareMode: true,
+                  hmr: {
+                    server,
+                  },
+                }
+              })
```

Comment:
I replaced `unstable_createViteServer` with direct use of `vite.createServer` since currently `unstable_createViteServer` doesn't support passing vite config.
It might make sense to tweak the API of `unstable_createViteServer`. Please let me know if anyone has opinion on this.

Also, FYI, using `server.hmr.server` option is a recommended setup from https://github.com/sapphi-red/vite-setup-catalogue/blob/48cde75352005aa1c1780f5eccf022db5619e285/examples/middleware-mode/server.js.

Reply:
> I replaced `unstable_createViteServer` with direct use of `vite.createServer` since currently `unstable_createViteServer` doesn't support passing vite config. It might make sense to tweak the API of `unstable_createViteServer` (or remove it entirely in favor of direct vite call?). Please let me know if anyone has opinion on this.

I think that direct use of `vite.createServer` is better, because if you have a need to create vite server it's because you need to customize something.

But if you want to have `unstable_createViteServer`, at least allow to pass config modifications. Current implementation is not doing something special (only setting `middlewareMode`) so it's not necessary.